src/visor/readable_test.go

```diff
@@ -1,343 +1,274 @@
+	}
+
+	sig := cipher.SignHash(gb.HashHeader(), s)
+	return sig
+}
 
 // Returns an appropriate VisorConfig and a master visor
-// func setupVisorConfig() (VisorConfig, *Visor) {
-// 	// Make a new master visor + blockchain
-// 	// Get the signed genesis block,
-// 	mw := wallet.NewWalletEntry()
-// 	mvc := NewVisorConfig()
-// 	mvc.CoinHourBurnFactor = 0
-// 	mvc.IsMaster = true
-// 	mvc.MasterKeys = mw
-// 	mvc.GenesisSignature = createGenesisSignature(mw)
-// 	mv := NewVisor(mvc)
-
-// 	// Use the master values for a client configuration
-// 	c := NewVisorConfig()
-// 	c.IsMaster = false
-// 	c.GenesisSignature = mvc.GenesisSignature
-// 	c.GenesisTimestamp = mvc.GenesisTimestamp
-// 	c.MasterKeys = mw
-// 	c.MasterKeys.Secret = cipher.SecKey{}
-// 	c.WalletDirectory = testWalletDir
-// 	return c, mv
-// }
-
-// func setupVisor() (v *Visor, mv *Visor) {
-// 	vc, mv := setupVisorConfig()
-// 	v = NewVisor(vc)
-// 	return
-// }
-
-// func setupVisorFromMaster(mv *Visor) *Visor {
-// 	vc := NewVisorConfig()
-// 	vc.IsMaster = false
-// 	vc.MasterKeys = mv.Config.MasterKeys
-// 	vc.MasterKeys.Secret = cipher.SecKey{}
-// 	vc.GenesisSignature = mv.blockSigs.Sigs[0]
-// 	vc.GenesisTimestamp = mv.blockchain.Blocks[0].Head.Time
-// 	return NewVisor(vc)
-// }
-
-// func setupMasterVisorConfig() VisorConfig {
-// 	// Create testmaster.keys file
-// 	c := NewVisorConfig()
-// 	c.CoinHourBurnFactor = 0
-// 	c.IsMaster = true
-// 	mw := wallet.NewWalletEntry()
-// 	c.MasterKeys = mw
-// 	c.GenesisSignature = createGenesisSignature(mw)
-// 	return c
-// }
-
-// func setupMasterVisor() *Visor {
-// 	return NewVisor(setupMasterVisorConfig())
-// }
-
-// func cleanupVisor() {
-// 	filenames := []string{
-// 		testMasterKeysFile,
-// 		testBlockchainFile,
-// 		testBlocksigsFile,
-// 		testWalletFile,
-// 		testWalletEntryFile,
-// 	}
-// 	for _, fn := range filenames {
-// 		os.Remove(fn)
-// 		os.Remove(fn + ".bak")
-// 		os.Remove(fn + ".tmp")
-// 	}
-// 	wallets, err := filepath.Glob("*." + wallet.WalletExt)
-// 	if err != nil {
-// 		logger.Critical("Failed to glob wallet files: %v", err)
-// 	} else {
-// 		for _, w := range wallets {
-// 			os.Remove(w)
-// 			os.Remove(w + ".bak")
-// 			os.Remove(w + ".tmp")
-// 		}
-// 	}
-// }
-
-func createUnconfirmedTxn(t *testing.T) UnconfirmedTxn {
-	ut := UnconfirmedTxn{}
-	ut.Txn = coin.Transaction{}
-	ut.Txn.InnerHash = testutil.RandSHA256(t)
-	ut.Received = utc.Now().UnixNano()
-	ut.Checked = ut.Received
-	ut.Announced = time.Time{}.UnixNano()
-	return ut
+func setupVisorConfig(t *testing.T) Config {
+	wltDir := prepareWltDir()
+	c := NewVisorConfig()
+	c.WalletDirectory = wltDir
+	c.GenesisSignature = createGenesisSignature(t)
+	return c
+}
+
+func setupVisor(t *testing.T) (v *Visor, close func()) {
+	db, close := testutil.PrepareDB(t)
+	vc := setupVisorConfig(t)
+	v, err := NewVisor(vc, db)
+	require.NoError(t, err)
+	return
+}
+
+func cleanupVisor(dir string) {
+	err := os.RemoveAll(dir)
```

Comment:
Don't do this, in case there is a bug and we call os.RemoveAll on the wrong directory (this happened before)

Just leave the temp folders in temp

Reply:
@gz-c , fixed.
Do I need remove wallet files?